### PR TITLE
Removing unitest in favor of pytest from test_config.py

### PR DESCRIPTION
**Pull request by rihp**
### Background

we are moving away from using the `unittests` library in favor of using pytest

### Changes

only changed the test_config.py test case
dropped unittest in favor of pytest

### Documentation

not necessary

### Test Plan

running `pytest` as usual
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -1,126 +1,132 @@
...
-        """
-        # Store debug mode to reset it after the test
-        debug_mode = self.config.debug_mode
-
-        self.config.set_debug_mode(True)
-        self.assertTrue(self.config.debug_mode)
-
-        # Reset debug mode
-        self.config.set_debug_mode(debug_mode)
+@pytest.fixture
```
Comment (collijk) [inline tests/test_config.py]: There's a fixture available in the main `conftest.py` so we don't need this.

File: tests/test_config.py [hunk 2]
```diff
@@ -1,126 +1,132 @@
...
+def config():
+    return Config()
+
+
+def test_singleton(config):
```
Comment (collijk) [inline tests/test_config.py]: We don't want this test.  It's not actually testing the config, it's testing the singleton metaclass

**Review by collijk — CHANGES_REQUESTED** [review]
Couple of small things, otherwise looking good!

File: tests/test_config.py (hunk 2 above)
Comment (rihp) [reply tests/test_config.py]: updated

File: tests/test_config.py (hunk 1 above)
Comment (rihp) [reply tests/test_config.py]: removed from this file

**Review by collijk — APPROVED** [review]

**Comment by AdelElo13** [comment]
Im am not subscribed to this why am i getijng a 100 notifications a day on email?

**Pull request merged**